controllers/front/listing/CategoryController.php

```diff
@@ -52,7 +52,22 @@ public function canonicalRedirection($canonicalURL = '')
 
     public function getCanonicalURL()
     {
-        return $this->context->link->getCategoryLink($this->category);
+        $canonicalUrl = $this->context->link->getCategoryLink($this->category);
+        $parsedUrl = parse_url($canonicalUrl);
+        if (isset($parsedUrl['query'])) {
+            parse_str($parsedUrl['query'], $params);
+        } else {
+            $params = array();
+        }
+        $page = (int) Tools::getValue('page');
+        if (false !== $page && $page > 1) {
```

Comment:
`false !== $page` is never true, because of the `(int)` you made the line before.